compiler/optimizer/VPHandlers.cpp

```diff
@@ -447,6 +446,25 @@ static bool tryFoldCompileTimeLoad(
                }
             return didSomething;
             }
+         else if (node->getSymbolReference() == vp->comp()->getSymRefTab()->findVftSymbolRef()
+                  && node->getFirstChild() == curNode
+                  && constraint->isNonNullObject()
```

Comment:
It seems like we should be able to do this transformation even if we don't have a non-null constraint - the object always has to be non-null when we actually load. I guess this condition is here to make sure that we don't mess up a `NULLCHK` that's combined with the load? We *could* transform in that case, but we'd have to separate the `NULLCHK`, which doesn't look to be easy to do here

Reply:
You are correct, `constraint->isNonNullObject()` is added to make sure, we do not transform indirect load of an object for which we need `NULLCHK` which as you mentioned, would be difficult to do here.